### DISCUSSION TITLE
Local storage - x-data showing old/wrong value after page refresh

### DISCUSSION THREAD

**Discussion by DarpanKulkarni** [alpinejs/alpine#1210] [1. Help] [answered]
I am trying to toggle dark mode saving the value in local storage. But when I toggle the dark mode on and off and refresh the page, the value in the local storage shows 'false' (which is expected) but the x-data sets the wrong value which is 'true'.

```
<!-- https://cdn.jsdelivr.net/gh/alpinejs/alpine@v2.8.2/dist/alpine.min.js -->
<body 
  x-data="{darkMode: localStorage.getItem('dark')}"
  x-init="$watch('darkMode', val => localStorage.setItem('dark', val))"
  x-bind:class="{ 'dark': darkMode }"
>
  
  <p x-show="darkMode">Dark Mode: ON</p>
  <p x-show="!darkMode">Dark Mode: OFF</p>
  
  <button
    @click="darkMode = !darkMode"
  >Toggle Dark Mode</button>
  
</body>
```

Check this fiddle for demo: https://jsfiddle.net/darpan_kulkarni/kpr5de0a

1. Turn on the dark mode
2. Turn off the dark mode
3. Refresh the page, it will still show dark mode ON

What I am doing wrong?

**Comment by SimoTod** [comment, marked as answer]
localStorage saves and retrieves everything as strings. The string `false` is a truthy value so it works as expected.
You need to change your x-data to `x-data="{darkMode: localStorage.getItem('dark') === 'true'}"`

**Reply by DarpanKulkarni** [reply]
Thank  you!

**Comment by ravindra135** [comment]
Thank You **@SimoTod** for mentioning the obvious problem;

However, this way I implemented dark mode with alpine.js

    In my app.js file
    
    const  systemTheme  =  window.matchMedia("(prefers-color-scheme: dark)").matches;
    //systemTheme returns 'boolean' type Value;
    
    let  darkTheme  =  localStorage.getItem('dark')
    //darkTheme returns 'string' type value
	
	if(darkTheme  ==  null) {
		darkTheme  =  systemTheme;
		/* 
		If user is opening site for very first time, 
		So, it will prefer going with System Settings.
		*/
	} else {
		darkTheme  = (/true/).test(localStorage.getItem('dark'));
		/*
		As, SimoTod mentioned localStorage will return string type, so we need to
		convert it into boolean value
		*/
	}
	
	Alpine.store('darkMode') {
		dark: darkTheme,
		toggle() {
			this.dark = ! this.dark;
			if(this.dark) {
				document.documentElement.classList.add('dark')
				//This will add 'dark' class to HTML;
				localStorage.setItem('dark', true)
			} else {
				document.documentElement.classList.remove('dark')
				localStorage.setItem('dark', false)
			}
		}
	}

Now, in Blade Template File;

    <html  lang="en" x-data x-cloak :class="$store.darkMode.dark ? 'dark' : '' ">

Now you can able to maintain the state of dark mode. **Cheers!!!**

**Reply by ekwoka** [reply]
Could use `JSON.parse`

Or just `=== true`
